pkg/controller/dynamodb/table/hooks.go

```diff
@@ -112,6 +119,38 @@ func postGenerateDeleteTableInput(cr *svcapitypes.Table, obj *svcsdk.DeleteTable
 	return obj
 }
 
+type tagger struct {
+	kube client.Client
+}
+
+func (e *tagger) Initialize(ctx context.Context, mg resource.Managed) error {
+	cr, ok := mg.(*svcapitypes.Table)
+	if !ok {
+		return errors.New(errUnexpectedObject)
+	}
+	tagMap := map[string]string{}
+	for _, t := range cr.Spec.ForProvider.Tags {
+		tagMap[aws.StringValue(t.Key)] = aws.StringValue(t.Value)
+	}
+	for k, v := range resource.GetExternalTags(cr) {
+		tagMap[k] = v
+	}
+	tags := make([]*svcapitypes.Tag, len(tagMap))
+	i := 0
+	for k, v := range tagMap {
+		tags[i] = &svcapitypes.Tag{Key: aws.String(k), Value: aws.String(v)}
+		i++
+	}
```

Comment:
```suggestion
	tags := make([]*svcapitypes.Tag, 0)
	for k, v := range tagMap {
		tags = append(tags, &svcapitypes.Tag{Key: aws.String(k), Value: aws.String(v)})
	}
```
Nitpick, but I think this looks a little more readable?